### PR TITLE
Syn 126 add new configuration options to simulator

### PR DESCRIPTION
**Pull request by Daresoul**
## Describe your changes
Changed to add configuaration options:
1. Maximum traces parameter
2. toggle for unique traces
3. Coverage insurance

There was a fourth namely: "Specify amount of traces" but sounded like the same as number 1. If i have misunderstood that, please leave it in a comment or the review.

## Checklist before requesting a review
- [ x ] I have performed a self-review of my code

### CODE REVIEW

**Review by Volume999 — APPROVED** [review]
lgtm

**Review by leny-mi — CHANGES_REQUESTED** [review]
The "number of traces" parameter directly specifies the number of traces we generate. The "max traces" parameter on the other hand specifies a maximum cutoff if we use coverage or other attributes to generate the traces. They are not the same, but rather two distinct termination conditions. 

They should also be distinguished, as it does not make sense to set a required coverage, and a specific amount of traces. The configuration needs to have a way to specify which termination condition we are using. 

You could do this using a discriminative union.

File: packages/synthchron-tauri/src/components/editor/rightSidebar/SimulationConfiguration.tsx [hunk 1]
```diff
@@ -34,6 +38,11 @@ const partialAutoConfiguration = {
   //These are values that can be automatically generated
   randomSeed: '',
   //Add other configuration options here
+  configurationName: 'Default',
+
+  terminationType: {
+    type: terminationType.Standard,
+  } as StandardConfigurationTerminationType,
```
Comment (leny-mi) [inline packages/synthchron-tauri/src/components/editor/rightSidebar/SimulationConfiguration.tsx:45]: Why is there a type cast?

File: packages/synthchron-simulator/src/types/simulationTypes.ts [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
   randomSeed: string
+  uniqueTraces?: boolean
+  terminationType:
+    | StandardConfigurationTerminationType
+    | CoverageTerminationType
+    | MaximumTracesTerminationType
+    | SpecifiedAmountOfTracesTerminationType
+}
+
+export enum terminationType {
```
Comment (leny-mi) [inline packages/synthchron-simulator/src/types/simulationTypes.ts:16]: Enums are written in PascalCase

File: packages/synthchron-simulator/src/types/simulationTypes.ts [hunk 3]
```diff
@@ -1,10 +1,43 @@
...
+}
+
+export type CoverageTerminationType = {
+  type: terminationType.Coverage
+  coverage: number
+}
+
+export type MaximumTracesTerminationType = {
+  type: terminationType.MaximumTraces
+  maximumTraces: number
```
Comment (leny-mi) [inline packages/synthchron-simulator/src/types/simulationTypes.ts:34]: The maximum traces parameter should not be a termination type, but rather a parameter. A user could always want to add a limit to the amount of traces generated, independent of termination type. This should therefore be part of the configuration.

**Review by leny-mi — CHANGES_REQUESTED** [review]
See comments. Mainly the maximum traces termination type should be updated. The rest is suggestions, but not required.

File: packages/synthchron-simulator/src/types/simulationTypes.ts (hunk 2 above)
Comment (Daresoul) [reply packages/synthchron-simulator/src/types/simulationTypes.ts:16]: done

File: packages/synthchron-simulator/src/types/simulationTypes.ts (hunk 3 above)
Comment (Daresoul) [reply packages/synthchron-simulator/src/types/simulationTypes.ts:34]: done

**Review by leny-mi — APPROVED** [review]
lgtm

**Pull request merged**